enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/HighAvailabilityMemberStateMachine.java

```diff
@@ -323,5 +357,32 @@ private long getTotalCount()
         {
             return Iterables.count( members.getMembers() );
         }
+
+        private boolean checkNewStateAndTransition( HighAvailabilityMemberState newState )
```

Comment:
i think we need to spilt it into 2 separate methods and keep assignment of new state outside of this method